spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceFilter.java

```diff
@@ -152,7 +152,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 		}
 		addToResponseIfNotPresent(response, Span.SAMPLED_NAME, skip ? Span.SPAN_NOT_SAMPLED : Span.SPAN_SAMPLED);
 		String name = HTTP_COMPONENT + ":" + uri;
-		spanFromRequest = createSpan(request, skip, spanFromRequest, name);
+		try {
+			spanFromRequest = createSpan(request, skip, spanFromRequest, name);
+		} catch (IllegalStateException e) {
+			filterChain.doFilter(request, response);
+			response.sendError(HttpStatus.BAD_REQUEST.value(), "Tracing data is malformed");
```

Comment:
I'd re-use the exception message here, as it could be written in such a way so that what to change is clear.

Ex. ` "Exception tracing request " + e.getMessage()`